modular_skyrat/master_files/code/modules/pain_shock/shock_helper.dm

```diff
@@ -0,0 +1,233 @@
+#define COMSIG_SHOCK_UPDATE "shock_update"
+
+/* /mob/living/carbon/Life(delta_time = SSMOBS_DT, times_fired)
+	flow_control() */
+/mob/living/carbon
+	var/lastpainmessage
+	var/in_shock = FALSE
+	var/flow_rate = BASE_FLOW_RATE
+	var/injuries
+/mob/living/carbon/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_SHOCK_UPDATE, .proc/shock_helper)
+
+/mob/living/carbon/proc/flow_control()
+	SIGNAL_HANDLER
+	injuries = calc_injuries()
+//		var/pain_score = injuries + length(all_wounds) // old, backup and sketchy
+	calc_pain()
+	if(stat == DEAD)
+		flow_rate = 0
+		update_health_hud()
+		return
+	else if(!in_shock)	flow_rate = clamp(rand(BASE_FLOW_RATE, BASE_FLOW_RATE_UPPER) + calc_pain(), 0, FLOW_RATE_ARREST)
+	in_shock ? shock_dying(flow_rate) : shock_helper(flow_rate)
+
+	update_health_hud()
+
+/mob/living/carbon/proc/calc_pain()
+	var/pain_score = 1
+	for(var/i in all_wounds)
+		var/datum/wound/iterwound = i
+		if(iterwound.severity == WOUND_SEVERITY_SEVERE || WOUND_SEVERITY_CRITICAL || WOUND_SEVERITY_LOSS)
+			pain_score += SHOCK_STAGE_MODERATE
+		if(iterwound.severity == WOUND_SEVERITY_MODERATE || WOUND_SEVERITY_TRIVIAL)
+			pain_score += SHOCK_STAGE_MINOR
+
+	pain_score += getBruteLoss() + getFireLoss() + getToxLoss() + getOxyLoss() + getStaminaLoss() / DAMAGE_TO_PAIN_DIVISION_FACTOR
+	return pain_score
+
+/mob/living/carbon/proc/calc_injuries()
+	var/injuries
+	for(var/i in all_wounds)
+		injuries++
+	return injuries
+
+/mob/living/carbon/proc/shock_dying(last_bpm, pulsetimer)
+	if(!in_shock || stat == DEAD)
+		return
+	if(can_leave_shock(last_bpm))
+		in_shock = FALSE
+		set_stat(CONSCIOUS)
+		to_chat(src, span_hypnophrase("You body tingles painfully as your nerves come back..."))
+	else if(!can_leave_shock(last_bpm))
+		if(stat != SOFT_CRIT)
+			set_stat(SOFT_CRIT)
+		current_pain_message_helper("Shock")
+		losebreath += 0.25
+	flow_rate -= losebreath // Double negative when in crit?
+
+	if(flow_rate <= 0 && stat != DEAD)
+		adjustOrganLoss(ORGAN_SLOT_BRAIN, losebreath)
+
+
+/mob/living/carbon/proc/can_leave_shock(last_bpm)
+	var/truepain = calc_pain()
+	if(truepain <= 100)
+		return TRUE
+	return FALSE
+/mob/living/carbon/proc/resetpainmsg()
+	lastpainmessage = null
+/mob/living/carbon/proc/current_pain_message_helper(current_pain)
+	if(lastpainmessage)
+		return
+	lastpainmessage = TRUE
+	addtimer(CALLBACK(src, .proc/resetpainmsg), 15 SECONDS)
+	var/list/close2death = list("a human", "a moth", "a felinid", "a lizard", "a particularly resilient slime", "a syndicate agent", "a clown", "a mime", "a mortal foe", "an innocent bystander")
+
+	switch(current_pain)
```

Comment:
should be turned into defines, otherwise it's hard to keep track of this mess of messages